### PR TITLE
Magento_Captcha: avoid using deprecated escape* methods from Abstract…

### PR DESCRIPTION
**Pull request by sergiynezbritskiy**
### Description (*)

Avoid using deprecated escape* methods from AbstractBlock

### Related Pull Requests

https://github.com/magento/magento2/pull/31610

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Captcha/view/adminhtml/templates/default.phtml [hunk 1]
```diff
@@ -7,24 +7,25 @@
 /** @var \Magento\Captcha\Block\Captcha\DefaultCaptcha $block */
 /** @var \Magento\Captcha\Model\DefaultModel $captcha */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
+/** @var \Magento\Framework\Escaper $escaper */
```
Comment (ihor-sviziev) [inline app/code/Magento/Captcha/view/adminhtml/templates/default.phtml]: Could you combine it to a single phpdoc block?

File: app/code/Magento/Captcha/view/frontend/templates/default.phtml [hunk 2]
```diff
@@ -5,29 +5,29 @@
  */
 
 /** @var \Magento\Captcha\Block\Captcha\DefaultCaptcha $block */
-
+/** @var \Magento\Framework\Escaper $escaper */
 /** @var \Magento\Captcha\Model\DefaultModel $captcha */
```
Comment (ihor-sviziev) [inline app/code/Magento/Captcha/view/frontend/templates/default.phtml]: Could you combine it to a single phpdoc block?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Captcha/view/adminhtml/templates/default.phtml (hunk 1 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Captcha/view/adminhtml/templates/default.phtml]: fixed

File: app/code/Magento/Captcha/view/frontend/templates/default.phtml (hunk 2 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Captcha/view/frontend/templates/default.phtml]: fixed

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Semantic Version Checker

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please resolve conflicts

**Comment by sergeynezbritskiy** [comment]
> Please resolve conflicts

Hello @ihor-sviziev, fixed

File: app/code/Magento/Captcha/view/frontend/templates/default.phtml (hunk 2 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Captcha/view/frontend/templates/default.phtml]: Trigger removing "Progress: needs update" lable

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Captcha/view/frontend/templates/default.phtml [hunk 3]
```diff
@@ -4,45 +4,49 @@
...
                                             "imageLoader": "<?= /* @noEscape */ $loaderUrl ?>",
-                                             "type": "<?= $block->escapeHtmlAttr($block->getFormId()) ?>"}}'>
+                                             "type": "<?= $escaper->escapeHtmlAttr($block->getFormId()) ?>"}}'>
                 <div class="control captcha-image">
-                    <img alt="<?= $block->escapeHtmlAttr(__('Please type the letters and numbers below')) ?>"
+                    <img alt="<?= $escaper->escapeHtmlAttr(__('Please type the letters and numbers below')) ?>"
                          class="captcha-img"
                          height="<?= /* @noEscape */ (float) $block->getImgHeight() ?>"
-                         src="<?= $block->escapeUrl($captcha->getImgSrc()) ?>"/>
+                         src="<?= $block->$escaper($captcha->getImgSrc()) ?>"/>
```
Comment (ihor-sviziev) [inline app/code/Magento/Captcha/view/frontend/templates/default.phtml]:
```suggestion
                         src="<?= $escaper->escapeUrl($captcha->getImgSrc()) ?>"/>
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Captcha/view/frontend/templates/default.phtml (hunk 3 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Captcha/view/frontend/templates/default.phtml]: fixed

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8910](https://jira.corp.magento.com/browse/ENGCOM-8910) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
@magento run Database Compare, Functional Tests EE

**Pull request closed without merging**